keystone_json_assignment/json.py

```diff
@@ -76,56 +78,68 @@ def _get_public_id(self, user_id):
             'local_id': user_id,
             'entity_type': 'user'})
 
+    def load_user_project_map(self):
+        mtime = datetime.datetime.utcfromtimestamp(
+            os.path.getmtime('/etc/keystone/user-project-map.json'))
+        if self.userprojectmap == {} or mtime > self.last_loaded:
+            with open('/etc/keystone/user-project-map.json', 'r') as f:
+                userprojectmap = yaml.load(f)
+                self.last_loaded = mtime
```

Comment:
Cosmetic: we could os fstat the fd here to save the a actual mtime that we loaded.

Reply:
To do that we would have to also move `if self.userprojectmap == {}` inside the `with open`